.github/workflows/ci-cd.yml

```diff
@@ -22,54 +22,91 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  CI-CD:
+  lint-check:
     runs-on: ubuntu-latest
     steps:
-      - name: Check out code
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - name: Setup Nodejs 16.x
-        uses: actions/setup-node@v3
-        with:
-          node-version: 16
-      - name: install
-        run: yarn --frozen-lockfile
-      # waiting for approval - uses: nrwl/nx-set-shas@v3
-      - name: set base branch
-        run: |
-          if [ $BRANCH_REF = 'refs/heads/main' ]; then 
-            echo 'BASE_BRANCH=origin/main~1' >> $GITHUB_ENV
-          else
-            echo 'BASE_BRANCH=origin/main' >> $GITHUB_ENV
-          fi
+      - uses: actions/checkout@v3
```

Comment:
Do we still need the `checkout` action since it's already used in `setup` action?

Reply:
@vincentsong `checkout` is required/mandatory before calling the actions.